crates/editor/src/element.rs

```diff
@@ -2062,21 +2062,27 @@ impl EditorElement {
                     None
                 };
 
-            let offset_range_start = snapshot
-                .display_point_to_anchor(DisplayPoint::new(range.start, 0), Bias::Left)
-                .to_offset(&snapshot.buffer_snapshot);
-            let offset_range_end = snapshot
-                .display_point_to_anchor(DisplayPoint::new(range.end, 0), Bias::Right)
-                .to_offset(&snapshot.buffer_snapshot);
+            let offset_range_start =
+                snapshot.display_point_to_anchor(DisplayPoint::new(range.start, 0), Bias::Left);
+            let offset_range_end =
+                snapshot.display_point_to_anchor(DisplayPoint::new(range.end, 0), Bias::Right);
 
             editor
                 .tasks
                 .iter()
                 .filter_map(|(_, tasks)| {
-                    if tasks.offset.0 < offset_range_start || tasks.offset.0 >= offset_range_end {
+                    if tasks
+                        .offset
+                        .cmp(&offset_range_start, &snapshot.buffer_snapshot)
+                        .is_lt()
+                        || tasks
+                            .offset
+                            .cmp(&offset_range_end, &snapshot.buffer_snapshot)
+                            .is_gt()
+                    {
                         return None;
                     }
-                    let multibuffer_point = tasks.offset.0.to_point(&snapshot.buffer_snapshot);
+                    let multibuffer_point = tasks.offset.to_point(&snapshot.buffer_snapshot);
```

Comment:
I recall futzing with this code block a while back because it showed up in profiles for mutli-buffers, which is where https://github.com/zed-industries/zed/pull/21849/files came from:
if I may suggest, maybe we could turn display range boundaries into points and then turn each anchor into point too before comparing them? That way we get to avoid two anchor comparisons.

Reply:
Interesting, I was also slightly skeptical of the linear search to find the range we're interested in, so maybe we should fix that too.

Reply:
I guess we can't unless we know what order the buffers are in in the excerpts, never mind.

Reply:
Yeah, we could probably do a double-partition-point thing to find the range efficiently.